src/features/session_replay/aggregate/index.js

```diff
@@ -180,12 +182,9 @@ export class Aggregate extends AggregateBase {
       this.mode = session.state.sessionReplayMode
     } else {
       // The session is new... determine the mode the new session should start in
-      if (fullSample) this.mode = MODE.FULL // full mode has precedence over error mode
-      else if (errorSample) this.mode = MODE.ERROR
-      // If neither are selected, then don't record (early return)
-      else {
-        return
-      }
+      this.mode = srMode
+      // If off, then don't record (early return)
+      if (this.mode === MODE.OFF) return
```

Comment:
```suggestion
      // If off, then don't record (early return)
      if ((this.mode = srMode) === MODE.OFF) return
```

Reply:
pointless nesting and harder to read tbh